packages/itwin/measure-tools/src/api/SheetMeasurementHelper.ts

```diff
@@ -159,16 +159,50 @@ export namespace SheetMeasurementsHelper {
-    if (!ev.viewport)
+  export function checkIfAllowedDrawingType(viewport: ScreenViewport | undefined, point: Point3d, allowedDrawingTypes: DrawingType[]) {
+    if (!viewport)
       return false;
-    for (const drawing of DrawingDataCache.getInstance().getSheetDrawingDataForViewport(ev.viewport)) {
+    for (const drawing of DrawingDataCache.getInstance().getSheetDrawingDataForViewport(viewport)) {
       if (allowedDrawingTypes.includes(drawing.type)) {
-        if (SheetMeasurementsHelper.checkIfInDrawing(ev.point, drawing.origin, drawing.extents)) {
+        if (SheetMeasurementsHelper.checkIfInDrawing(point, drawing.origin, drawing.extents)) {
           return true;
         }
       }
     }
     return false;
   }
+
+  function getNameFromDrawingType(type: SheetMeasurementsHelper.DrawingType): string  {
+    if (type === SheetMeasurementsHelper.DrawingType.CrossSection) {
+      return IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MeasurementNames.CrossSection");
+    } else if (type === SheetMeasurementsHelper.DrawingType.Plan) {
+      return IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MeasurementNames.Plan");
+    } else if (type === SheetMeasurementsHelper.DrawingType.Profile) {
+      return IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MeasurementNames.Profile");
+    } else {
+      return "";
+    }
+  }
+
+  export async function getSheetErrorToolTipText(hit: HitDetail, allowedDrawingTypesList: SheetMeasurementsHelper.DrawingType[], defaultToolTip:(hit: HitDetail) => Promise<HTMLElement | string>): Promise<string | HTMLElement> {
+    if (SheetMeasurementsHelper.checkIfAllowedDrawingType(hit.viewport, hit.hitPoint, allowedDrawingTypesList)) {
+      return defaultToolTip(hit);
+    }
+
+    if (allowedDrawingTypesList.length < 1) {
+      return "";
+    }
+    if (allowedDrawingTypesList.length === 1) {
+      return IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.OneMeasurementInvalid", { drawingName: getNameFromDrawingType(allowedDrawingTypesList[0])});
+    }
+    if (allowedDrawingTypesList.length > 1){
+      let result = IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MoreMeasurementsInvalidHead", { drawingName: getNameFromDrawingType(allowedDrawingTypesList[0])});
+      for (let i = 1; i < allowedDrawingTypesList.length - 1; i++) {
+        result = result + IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MoreMeasurementsInvalidMiddle", { drawingName: getNameFromDrawingType(allowedDrawingTypesList[i])});
+      }
+      result = result + IModelApp.localization.getLocalizedString("CivilReviewTools:SheetMeasurementTooltip.MoreMeasurementsInvalidLast", { drawingName: getNameFromDrawingType(allowedDrawingTypesList[allowedDrawingTypesList.length - 1])});
+      return result;
+    }
+    return "";
```

Comment:
For code clarity, I'd have the final return be your `length===1` case, since you had already checked if zero or if greater than one.